src/libertem/udf/base.py

```diff
@@ -2111,12 +2119,15 @@ def _run_tile(
         roi: Optional[np.ndarray],
     ) -> None:
         for udf in udfs:
-            if isinstance(udf, UDFTileMixin):
+            udf_method = udf.get_method()
+            if udf_method == UDFMethodEnum.TILE:
+                assert isinstance(udf, UDFTileMixin)  # mypy!
```

Comment:
As an alternative, we can do something like this:

```python
                udf_tiled = typing.cast(UDFTileMixin, udf)
                udf_tiled.process_tile(device_tile)
```

(keeping the other method accesses like `udf.set_slice` as they are)

`cast` should be the identity function at runtime. Not the most elegant way (we don't actually check the type, we just say: trust me, basically), but maybe something to consider if we want to keep overheads low.

Reply:
That's a good solution, thanks for the suggestion. (I didn't know about `cast`).

On the subject of overheads, we could remove one `isinstance` call per tile (in the default/standard case) by moving the `get_method()` call one level up into _`run_udfs` and reusing the result for each `_run_tile`. Do you want me to do that, I think it would be consequence-free, or is it not worth the bother?

Reply:
> On the subject of overheads, we could remove one isinstance call per-UDF per-tile

Sounds good to me! Also good if people do whatever in their `get_method` impl...

Reply:
It's not elegant, but it's done - could probably be merged with the execution_plan or better still frozen during `_prepare_run_for_dataset` : https://github.com/LiberTEM/LiberTEM/pull/1509/commits/ccc9f3af7416e52a5725fe1d803019fb5714a93b